File: src/main/java/com/procurement/evaluation/model/dto/ocds/OrganizationReference.java
package com.procurement.evaluation.model.dto.ocds;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.Getter;
import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

@Getter
@JsonPropertyOrder({
    "id",
    "name"
})
public class OrganizationReference {

    @NotNull
    @JsonProperty("id")
    private final String id;

    @NotNull
    @Size(min = 1)
    @JsonProperty("name")
    private final String name;

    @JsonCreator
    public OrganizationReference(@JsonProperty("name") final String name,
                                 @JsonProperty("id") final String id) {
        this.id = id;
        this.name = name;
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder().append(name)
                                    .append(id)
                                    .toHashCode();
    }

    @Override
    public boolean equals(final Object other) {
        if (other == this) {
            return true;
        }
        if (!(other instanceof OrganizationReference)) {
            return false;
        }
        final OrganizationReference rhs = (OrganizationReference) other;
        return new EqualsBuilder().append(name, rhs.name)
                                  .append(id, rhs.id)
                                  .isEquals();
    }
}

File: src/main/kotlin/com/procurement/evaluation/model/dto/ocds/OrganizationReference.kt
package com.procurement.evaluation.model.dto.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import java.time.LocalDateTime

data class OrganizationReference @JsonCreator constructor(

    val id: String,

    val name: String,

    val identifier: Identifier,

    val address: Address,

    val contactPoint: ContactPoint,
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    val additionalIdentifiers: MutableList<Identifier>?,

    @JsonInclude(JsonInclude.Include.NON_NULL)
    var details: Details?,

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    val persones: List<Person>?
) {
    data class Person(
        val title: String,
        val name: String,
        val identifier: Identifier,
        val businessFunctions: List<BusinessFunction>
    ) {
        data class Identifier(
            val scheme: String,
            val id: String,

            @JsonInclude(JsonInclude.Include.NON_NULL)
            val uri: String?
        )

        data class BusinessFunction(
            val id: String,
            val type: BusinessFunctionType,
            val jobTitle: String,
            val period: Period,

            @JsonInclude(JsonInclude.Include.NON_EMPTY)
            val documents: List<Document>?
        ) {
            data class Period(
                val startDate: LocalDateTime
            )

            data class Document(
                val id: String,
                val documentType: String,
                val title: String,

                @JsonInclude(JsonInclude.Include.NON_NULL)
                val description: String?
            )
        }
    }
}
